event_name_factory_test.go

```diff
@@ -0,0 +1,33 @@
+/*
+ * Flow Go SDK
+ *
+ * Copyright 2019 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package flow
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEventNameFactory(t *testing.T) {
+	assert.Equal(t, "A.7e60df042a9c0868.FlowToken.AccountCreated", NewEvent().
```

Comment:
I noticed you took a slightly different approach with the chaining than on the original issue. Can you explain why you chose this approach over the other?

Reply:
This pattern actually does seem to fit the patterns in the rest of the SDK. Thoughts on changing `.Build` to `.String` for more flexibility with the event struct later though?

Reply:
updated here https://github.com/onflow/flow-go-sdk/pull/449/commits/155a11e76499d84ef5487b8d1ca7cd29d57cf428.

@chasefleming I'd be happy to propose the original design if you fancy ;)